### PR TITLE
feat: new theme package to show all the tags as a collection

### PR DESCRIPTION
**Pull request by sawilde**
new theme package to show all the tags as a collection of pages, similar to how posts are displayed in the main page

1. TagsIcon - currently this is a copy of the twitter icon - advice needed on a good place to source svg icons that won't get this project in any strife due to attribution requirements.
2. TagsHeader - this is supplied so that people can customise the header ie add a banner image or similar
3. I used google translate for the translations - probably garbage
4. Updated SiteNav to host a dummy ShowTags that can be overwritten by the package
5. TagCard will also show an image for the tag if it is configured in the ghost CMS

I am not keen on how I have hooked in the tags page - I hope there is a better way.

I have been using this code on my own site for about a year and had no issues - only just converted to package though so feedback appreciated as it is probably a little rough for general consumption.

### CODE REVIEW

**Comment by sawilde** [comment]
TagIcon - implemented the same way as TwitterIcon (in fact it is just a copy of, as I don't have a good source of free svg icons.) - this needs some tinkering - the alternative is to add it to the menu like the contact plugin - I am not sure there is a place in Ghost to configure this.

TagHeader -  currently the data is in a localisation file - which is why I created that piece as a separate component to allow it to be overridden. This is the first pass at turning it into a plugin and it may be possible to pass in the config for the title via the options - again not sure if such a place in Ghost exists for this.

tags.js - currently this is a page (as that is how I currently host it) but looking at the other plugins I think I may need to inject it as a PageNode (similar to contact) but I need to get my head on how that works.

All the other information for the tags (title, image, description, count) comes from the ghost CMS itself

**Comment by sawilde** [comment]
@styxlab - some updates

tags.js is now a template and is being added as a page (pageNode) - title and path pulled in via config - ditched the dodgy language translations (I assume people can now configure this in the config)

TagIcon is gone and I am now using FontAwesome and using the faTags icon (the attribution is in the svg and so good)

Renamed some of the components

File: packages/gatsby-theme-ghost-tags/README.md
Comment (styxlab) [inline packages/gatsby-theme-ghost-tags/README.md:1]: Please rename to gatsby-theme-ghost-tags

**Review by styxlab — CHANGES_REQUESTED** [review]
First review comments (without testing the code yet)
- Looks very good overall, I think this is a great addtion to this package
- You can link to yourself, where it make sense (contributor, author. etc.)
- You can also add yourself as a contributor  to main READMEs (theme and starter)
- .babelrc: This could potentially mess up a lot of things and I didn't seem to need it before. Why did you include those presets?
- I am not sure if everyone want the social icon, some may prefer a navigation link: could the social icon part be made optional? I leave it to you to decide.

I will perform more tests after the next update.

Thanks again, this is a substantial contribution!

File: packages/gatsby-theme-ghost-tags/package.json
Comment (sawilde) [reply packages/gatsby-theme-ghost-tags/package.json:6]: can do - is this the github user or the npm user?

File: packages/gatsby-theme-ghost-tags/.babelrc
Comment (sawilde) [reply packages/gatsby-theme-ghost-tags/.babelrc:1]: good catch - not sure - will look into it

File: packages/gatsby-theme-ghost-tags/README.md
Comment (sawilde) [reply packages/gatsby-theme-ghost-tags/README.md:1]: good catch - I thought I had found them all in my cloning

File: packages/gatsby-theme-ghost-tags/.babelrc
Comment (sawilde) [reply packages/gatsby-theme-ghost-tags/.babelrc:1]: both the `contact` and `members` plugins have this file and I copied it across when I cloned (I never checked as to why it was there)

**Comment by sawilde** [comment]
@styxlab added ability control Icon visibility - I haven't added it as a navigation entry yet 
2 ways  
1. add tags page to your Ghost Blog that points to /tags/ (this is how I do it the moment)
2. add something similar to how the `contacts` plugin appends its entry to the nav

File: packages/gatsby-theme-ghost-tags/README.md
Comment (sawilde) [reply packages/gatsby-theme-ghost-tags/README.md:25]: done

**Comment by sawilde** [comment]
adding it to the navigation menu (as contact does - or perhaps doesn't) will take time.

I tried disabling the hook for contact by updating the setting as `navigation: []` and the contact menu entry remained

File: packages/gatsby-theme-ghost-tags/.babelrc
Comment (styxlab) [reply packages/gatsby-theme-ghost-tags/.babelrc:1]: okay, that's fine.

File: packages/gatsby-theme-ghost-tags/package.json
Comment (styxlab) [reply packages/gatsby-theme-ghost-tags/package.json:6]: You are correct, author can only be one person and it needs to be the npm user. But you can add a contributors field: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#people-fields-author-contributors

**Comment by styxlab** [comment]
- Things look fine from my side
- no need for special `navigation: []` handling as the endpoint `/tags` can be configured in Ghost to display a navigation link.
- I'll let you do final polishing, before I'll merge and publish

**Review by styxlab — CHANGES_REQUESTED** [review]
- Waiting for your final changes (if any).

**Comment by sawilde** [comment]
I think I have done what I needed to for this

**Review by styxlab — APPROVED** [review]

**Comment by styxlab** [comment]
Please be patient with me as I need a couple of days before I can do my final tests, merge and publish.

**Comment by sawilde** [comment]
no worries - how do you publish the packages once they have merged? Do you have a separate build pipeline?

**Comment by styxlab** [comment]
Thanks so much @sawilde ❤️!!!

**Comment by styxlab** [comment]
This package still needs to be published to npm....

**Comment by styxlab** [comment]
Finally published on npm. Sorry for the dealy and thanks again for your contributions!

**Comment by sawilde** [comment]
thanks and it was a pleasure

**Pull request merged**
